bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/ManagedThingProvider.java

```diff
@@ -47,4 +66,37 @@ protected String getStorageName() {
     protected String keyToString(ThingUID key) {
         return key.toString();
     }
+
+    @Override
+    protected @Nullable Thing toElement(String key, ThingDTO persistableElement) {
+        if (this.things.containsKey(key)) {
+            Thing thing;
+            thing = this.things.get(key);
+            return thing;
+        } else {
+            ThingTypeUID thingTypeUID = new ThingTypeUID(persistableElement.thingTypeUID);
+            boolean isBridge = thingTypeRegistry.getThingType(thingTypeUID) instanceof BridgeType;
+
+            Thing thing = ThingDTOMapper.map(persistableElement, isBridge);
+
+            // Add the thing handler and thing status
+            ThingHandler thingHandler = thingManager.getHandler(thing.getUID());
```

Comment:
Could you briefly explain why we need to get and set the `ThingHandler` and the `ThingStatusInfo` after restoring a thing? IIRC this was not necessary before because the `DefaultAbstractManagedProvider` just returns the plain element from its storage.

Reply:
I wasn't aware of the references in the <code>ThingRegistry</code>. I thought that the <code>ThingRegistry</code> was fetching from their providers on every request. So therefore the <code>ManagedThingProvider</code> was providing the complete <code>Thing</code>. As a consequence, the <code>ManagedThingProvider</code> now provides the part available in storage of the <code>Thing</code> and <code>ThingHandler</code> and <code>ThingStatusInfor</code> would be <code>null</code> if fetched directly from the <code>ManagedThingProvider</code>.